src/components/order/OrderFilter.js

```diff
@@ -87,50 +89,46 @@ class Filter extends PureComponent {
+            this.setState({sortOrder: isAdmin(this.props.user) ? "+" : "-"});
         }
     }
 
-    getCart = (filterState, next, searchQuery) => {
+    getCart = (filterState, next, searchQuery, isToggleSort) => {
         if (filterState === -1)
             return [];
-            
-        let queryparam = (next === true 
-            ? this.state.nextPageUrl 
+
+        let queryparam = (next === true
+            ? this.state.nextPageUrl
             : (next === undefined ? this.defaultPageUrl : this.state.prevPageUrl));
-        
+
         if (filterState !== '-10')
             queryparam += '&status=' + filterState;
 
-        if (searchQuery) 
+        if (searchQuery)
             queryparam += '&' + getQueryFromJson(searchQuery);
-        
+
         queryparam += '&sort=';
-        if (isAdmin(this.props.user)) {
-            switch (filterState) {                
-                case rcartStatus.placed:
-                    queryparam += '+statusChangeTime.placed.time'; break;
-                case rcartStatus.processing:
-                    queryparam += '+statusChangeTime.processing.time'; break;
-                case rcartStatus.readyToPickup:
-                    queryparam += '+statusChangeTime.readyToPickup.time'; break;
-                case rcartStatus.readyToDeliver:
-                    queryparam += '+statusChangeTime.readyToDeliver.time'; break;
-                default:
-                    queryparam += '-statusChangeTime.placed.time';
-            }
-        } else if (isStudent(this.props.user)) {
-            switch(filterState) {
-                case rcartStatus.invalid:
-                    queryparam += '-statusChangeTime.invalid.time'; break;
-                case rcartStatus.paymentFailed:
-                    queryparam += '-statusChangeTime.paymentFailed.time'; break;
-                case rcartStatus.processingPayment:
-                    queryparam += '-statusChangeTime.processingPayment.time'; break;
-                default:
-                    queryparam += '-statusChangeTime.placed.time';
-            }
+        let sortOrder = this.state.sortOrder;
+        if(isToggleSort){
+            sortOrder = sortOrder==="+"?"-":"+";
         }
-            
+        switch (filterState) {
+            case rcartStatus.placed:
+                queryparam += sortOrder + 'statusChangeTime.placed.time';
+                break;
+            case rcartStatus.processing:
+                queryparam += sortOrder + 'statusChangeTime.processing.time';
+                break;
+            case rcartStatus.readyToPickup:
+                queryparam += sortOrder + 'statusChangeTime.readyToPickup.time';
+                break;
+            case rcartStatus.readyToDeliver:
+                queryparam += sortOrder + 'statusChangeTime.readyToDeliver.time';
+                break;
+            default:
+                queryparam += sortOrder + 'statusChangeTime.placed.time';
```

Comment:
This will not work in the case of PaymentFailed and ProcessingPayment status filters as Placed time would not have been set. 
_Suggestion: Set default search method using previous switch block and change accordingly whenever the sort button is clicked._